http-clients-api/src/main/java/com/palantir/remoting/clients/ClientConfig.java

```diff
@@ -18,129 +18,81 @@
-// TODO(rfink) Use immutables? How is this different from ServiceConfiguration? Is it just the SslConfiguration?
-public final class ClientConfig {
+@Value.Immutable
+@SuppressWarnings("checkstyle:designforextension")
+public abstract class ClientConfig {
 
-    private static final Duration CONNECT_TIMEOUT = Duration.standardSeconds(10);
-    private static final Duration READ_TIMEOUT = Duration.standardMinutes(10);
-    private static final Duration WRITE_TIMEOUT = Duration.standardMinutes(10);
+    private static final Duration CONNECT_TIMEOUT = Duration.seconds(10);
+    private static final Duration READ_TIMEOUT = Duration.minutes(10);
+    private static final Duration WRITE_TIMEOUT = Duration.minutes(10);
     private static final int MAX_NUM_RETRIES = 1;
 
-    private Optional<SSLSocketFactory> thisSslSocketFactory = Optional.absent();
-    private Optional<X509TrustManager> thisTrustManager = Optional.absent();
-    private Optional<Duration> thisConnectTimeout = Optional.absent();
-    private Optional<Duration> thisReadTimeout = Optional.absent();
-    private Optional<Duration> thisWriteTimeout = Optional.absent();
-    private Optional<ProxyConfiguration> thisProxyConfiguration = Optional.absent();
-    private Optional<Integer> thisMaxNumRetries = Optional.absent();
+    @Value.Parameter
+    public abstract Optional<SSLSocketFactory> sslSocketFactory();
 
+    @Value.Parameter
+    public abstract Optional<X509TrustManager> trustManager();
 
-    public Optional<SSLSocketFactory> getSslSocketFactory() {
-        return thisSslSocketFactory;
+    @Value.Default
+    public Duration connectTimeout() {
+        return CONNECT_TIMEOUT;
     }
 
-    public Optional<X509TrustManager> getX509TrustManager() {
-        return thisTrustManager;
+    @Value.Default
+    public Duration readTimeout() {
+        return READ_TIMEOUT;
     }
 
-    public Duration getConnectTimeout() {
-        return thisConnectTimeout.or(CONNECT_TIMEOUT);
+    @Value.Default
+    public Duration writeTimeout() {
+        return WRITE_TIMEOUT;
     }
 
-    public Duration getReadTimeout() {
-        return thisReadTimeout.or(READ_TIMEOUT);
-    }
-
-    public Duration getWriteTimeout() {
-        return thisWriteTimeout.or(WRITE_TIMEOUT);
-    }
-
-    public int getMaxNumRetries() {
-        return thisMaxNumRetries.or(MAX_NUM_RETRIES);
-    }
+    @Value.Parameter
+    public abstract Optional<ProxyConfiguration> proxy();
 
-    public Optional<ProxyConfiguration> getProxyConfiguration() {
-        return thisProxyConfiguration;
+    @Value.Default
+    public Integer maxNumRetries() {
+        return MAX_NUM_RETRIES;
     }
 
-    public static ClientConfig empty() {
-        return new ClientConfig();
+    @Value.Check
+    public void check() {
+        Preconditions.checkState(sslSocketFactory().isPresent() == trustManager().isPresent(),
+                "Must set either both sslSocketFactory and TrustManager, or neither");
```

Comment:
should either read: "Must set either both SslSocketFactory and TrustManager, or neither" or "Must set either both sslSocketFactory and trustManager, or neither"